Competitive_Programming/Java/Stacks & Queues/Valid Brackets/Valid-Brackets-Using-Stack.md

```diff
@@ -0,0 +1,168 @@
+<h1 align = "center"><strong>Valid Brackets Using Stack</strong></h1>
+<hr>
+<h2>Problem Statement :- <br><br><em><strong> • Check for Balanced Brackets in an expression using Stack.</strong></em></h2>
+<hr>
+<h2><strong>Brackets which we all most commonly use or encounter are of 3 main types :-</strong></h2>
+<ul>
+    <li>Round brackets, open brackets or parentheses: ( )</li>
+    <li>Square brackets, closed brackets or box brackets: [ ]</li>
+    <li>curly brackets, squiggly brackets, swirly brackets, braces, or chicken lips: { }</li>
+</ul>
+<br>
+<h2><strong>Let's Understand the approach first :-</strong></h2>
+<p>
+<h3>We have to determine whether the given input of Brackets are valid or not.</h3>
+<ul>
+<br>
+<li>To check this we first use stack data structure, which works on a principle of <strong>LIFO (Last in first out) .</strong></li><br>
+
+<li>We start iterating over the input string given to us , and in the if clause (statement) whenever we encounter a open bracket i.e. <strong> ( , [ , { </strong> we push that bracket in the stack data structure which we have used.</li>
+<br>
+
+<li>And, on the other hand in the else clause (condition).</li>
+<br>
+<ul>
+    <li>We first always check that our stack should not be empty , because if the stack is empty that implies that the first character we encountered in the string is a close bracket. and if we get this condition we just simply need to return back from the function and no need to check further ,<strong> hence the given input of strings are not Valid Brackets.</strong></li><br>
```

Comment:
```suggestion
    <li>We first always check that our stack should not be empty because if the stack is empty then that implies that the first character we encountered in the string is a close bracket. If we get this condition we just simply need to return back from the function and no need to check further,<strong> hence the given input of strings are not Valid Brackets.</strong></li><br>
```